src/vs/editor/contrib/snippet/snippetVariables.ts

```diff
@@ -170,3 +177,29 @@ export class ClipboardBasedVariableResolver implements VariableResolver {
 		}
 	}
 }
+
+export class TimeBasedVariableResolver implements VariableResolver {
+
+	resolve(variable: Variable): string {
+		const { name } = variable;
+		const zeroPad = (n: string): string => n.length < 2 ? `0${n}` : n;
```

Comment:
better use the existing util from base/common/strings#pad

Reply:
Didn't know about it